### PR TITLE
Finalize refactor

### PR DESCRIPTION
**Pull request by paulfantom**
(Possibly) a final part of #566.

Doing few things in one PR:
- Breaking up `mixins/monitoring-mixins.libsonnet` into smaller chunks and assign those into own components.
- created `mixins` directory for mixins not tied to any component (like kubernetes-mixins) and moved `alerts/` and `rules/` there
- refactored grafana inclusion.
- "helmized" configuration. Now we have `values` map with all basic configuration entries. For helm users, this should look familiar and reduce the barrier of entry.

I decided to do final works in one PR to reduce number of rebases I would need to go through.

There are at least 2 follow-up action items:
- refactor brancz/kubernetes-grafana to be easier to include
- traverse object tree and extract dashboards from mixins to pass to grafana.

### CODE REVIEW

**Comment by paulfantom** [comment]
Ready for review. Ping @prometheus-operator/kube-prometheus-reviewers

**Comment by paulfantom** [comment]
After this, I would like to focus on refactoring the directory layout (issue https://github.com/prometheus-operator/kube-prometheus/issues/874) as the current one makes it quite hard to discover offered features by new users.

File: jsonnet/kube-prometheus/grafana/grafana.libsonnet [hunk 1]
```diff
@@ -0,0 +1,85 @@
+local defaults = {
```
Comment (brancz) [inline jsonnet/kube-prometheus/grafana/grafana.libsonnet:1]: any reason why we wouldn't just refactor kubernetes-grafana in the same way?

**Review by brancz — APPROVED** [review]
_Very_ nice!

File: jsonnet/kube-prometheus/grafana/grafana.libsonnet (hunk 1 above)
Comment (paulfantom) [reply jsonnet/kube-prometheus/grafana/grafana.libsonnet:1]: I want to do this. But this PR as is allows me to parallelize work on kube-prometheus and kubernetes-grafana.

Comment (brancz) [reply jsonnet/kube-prometheus/grafana/grafana.libsonnet:1]: Sounds very reasonable. Nice work!

**Pull request merged**